Cookbook/Technical-Documents/AutomateAUIFlow.md

```diff
@@ -0,0 +1,259 @@
+                  Cell, 0x600000d9dce0, {{0.0, 187.7}, {375.0, 71.0}}
+                    TextField, 0x600000d9ddc0, {{32.0, 224.0}, {271.0, 21.0}}, identifier: 'email', value: ukautomationuser@e...
+                  Cell, 0x600000d9dea0, {{0.0, 258.7}, {375.0, 71.0}}
+                    TextField, 0x600000d9df80, {{32.0, 295.0}, {271.0, 21.0}}, identifier: 'password', value: Pa55word
+                  Cell, 0x600000d9e060, {{0.0, 329.7}, {375.0, 64.0}}
+                    Button, 0x600000d9e140, {{0.0, 329.7}, {343.0, 48.0}}, identifier: 'loginButton', label: 'Log in'
+                    Other, 0x600000d9e220, {{0.0, 329.7}, {375.0, 64.0}}
+                      Other, 0x600000d9e300, {{16.0, 337.7}, {343.0, 48.0}}
+                        Other, 0x600000d9e3e0, {{16.0, 337.7}, {343.0, 48.0}}
+                          Button, 0x600000d9e4c0, {{16.0, 337.7}, {343.0, 48.0}}, identifier: 'loginButton', label: 'Log in'
+
+```
+
+For our test scenario we need to located three elements with each to have there own function, two to enter text and one to press a button. We already defined these functions, with two taking a String as a parameter. 
+
+#### Finding A Element
+
+Before beginning a more detail explanation of find element is [here](https://github.com/Babylonpartners/ios-playbook/blob/030750054285d8a21b562019053b40dbe56fc47e/Cookbook/Technical-Documents/AutomationIdentifiers.md#how-to-find-elements). 
+
+But for the purpose of this article we are trying to find the element as fast and simply as possible. To begin with we define a base query, which will reduces the search time. Looking at the view we can the *XCUIElement* is contained within a cell, which is within a table. So we use the base query `app.tables.cells`.
+
+Once you have the base query now you need an accessibility identifier, we will store these as private enums in the screen object. Though on new screen we can use the **AccessibilityContent** method.
+
+``` swift
+    fileprivate enum TextFields {
+        static let email = "Email"
+        static let password = "Password"
+    }
+```
+
+Then we search for the specific *XCUIElement*, the most common and readable method is ```app.tables.cells.textFields[TextFields.email]``` which will find the XCUIElement, but has potential drawbacks. The framework will search the entire view hierarchy for matches and throw an exception if multiple matches are found. The exception makes sense as the return type is `XCUIElement` and not `XCUIElementQuery`.
+
+An alternate solution or for cases were you want to find an element and know that there are potentially multiple matches, I would suggest using a `XCUIElementQuery` combined with `.firstMatch`. Giving us alternate code for locating the element, ```app.tables.cells.textFields.matching(identifier: TextFields.email).firstMatch```. On thing to note with `.firstMatch` is that it will stop the query on the first match, making the query technically faster. 
```

Comment:
```suggestion
An alternate solution or for cases were you want to find an element and know that there are potentially multiple matches, I would suggest using a `XCUIElementQuery` combined with `.firstMatch`. Giving us alternate code for locating the element, ```app.tables.cells.textFields.matching(identifier: TextFields.email).firstMatch```. One thing to note with `.firstMatch` is that it will stop the query on the first match, making the query technically faster. 
```